File: practice/algorithms/implementation/mini-max_sum/MinMaxSum.js
'use strict';

const processData = input => {
    const numbers = input.split(" ").map(i => parseInt(i));
    numbers.sort((a, b) => a - b)
    let minSum = 0;
    let maxSum = 0;
    for(let i = 0; i < numbers.length - 1; i++) {
        minSum += numbers[i];
        maxSum += numbers[i + 1];
    }
    console.log(minSum + " " + maxSum);
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/implementation/mini-max_sum/min_max_sum.rb
nums = gets.strip.split(" ").map(&:to_i)
nums.sort! {|a, b| a <=> b}


min_sum = 0
max_sum = 0
for i in (0...(nums.length - 1)) do
  min_sum += nums[i]
  max_sum += nums[i + 1]
end


puts "" + min_sum.to_s + " " + max_sum.to_s